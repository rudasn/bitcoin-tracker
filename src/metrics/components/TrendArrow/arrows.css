.arrow {
  width: 1.2rem;
  height: 1.2rem;
  margin: 0 0.5rem;
  display: inline-block;
  border-bottom: 5px solid var(--color-muted);
}
.arrow-down {
  width: 0;
  height: 0;
  border-left: 1.2rem solid transparent;
  border-right: 1.2rem solid transparent;
  border-top: 1.2rem solid var(--color-danger);
  border-bottom: none;
}
.arrow-up {
  width: 0;
  height: 0;
  border-left: 1.2rem solid transparent;
  border-right: 1.2rem solid transparent;
  border-bottom: 1.2rem solid var(--color-success);
}
