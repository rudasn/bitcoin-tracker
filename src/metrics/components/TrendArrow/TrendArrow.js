import React from 'react'

import './arrows.css'
import ChangeColoredText from '../ChangeColoredText'

export const Arrow = ({ change=0 }={}) => {
  if (change > 0) {
    return <span className="arrow arrow-up" />
  } else if (change < 0) {
    return <span className="arrow arrow-down" />
  }
  return <span className="arrow" />
}

export default ({ change=0, }) =>
  <ChangeColoredText change={ change }>
    <Arrow change={ change } />
  </ChangeColoredText>
