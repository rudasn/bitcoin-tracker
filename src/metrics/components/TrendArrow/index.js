import TrendArrow from './TrendArrow'

export default TrendArrow
