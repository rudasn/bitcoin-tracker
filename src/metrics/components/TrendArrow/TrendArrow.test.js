import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import { Arrow } from './TrendArrow'

it('renders without crashing', () => {
  const element = <Arrow />
  const div = document.createElement('div')
  ReactDOM.render(element, div)
  ReactDOM.unmountComponentAtNode(div)
})

it('renders no change correctly', () => {
  const element = <Arrow change={ 0 } />
  const tree = renderer.create(element).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders positive correctly', () => {
  const element = <Arrow change={ 1 } />
  const tree = renderer.create(element).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders negative correctly', () => {
  const element = <Arrow change={ -1 } />
  const tree = renderer.create(element).toJSON()
  expect(tree).toMatchSnapshot()
})
