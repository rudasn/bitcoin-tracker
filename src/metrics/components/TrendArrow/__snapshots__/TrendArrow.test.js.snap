// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders negative correctly 1`] = `
<span
  className="arrow arrow-down"
/>
`;

exports[`renders no change correctly 1`] = `
<span
  className="arrow"
/>
`;

exports[`renders positive correctly 1`] = `
<span
  className="arrow arrow-up"
/>
`;
